lib/pusher-platform/error_with_status.rb

```diff
@@ -0,0 +1,16 @@
+require_relative './common'
+
+module PusherPlatform
+  class ErrorWithStatus < Error
```

Comment:
Why do we need another error type?

Reply:
The `ErrorResponse` error fit with the case where you make a request and get back error info in the response (description, error type, etc).

This type is about returning an error with a suggested status code. It's only used by auth-related things so perhaps the error could be named differently: `AuthenticationError(WithStatus)`?